stripe.go

```diff
@@ -485,3 +517,44 @@ func initUserAgent() {
 	}
 	encodedStripeUserAgent = string(marshaled)
 }
+
+// Checks if an error is a problem that we should retry on. This includes both
+// socket errors that may represent an intermittent problem and some special
+// HTTP statuses.
+func (s *BackendConfiguration) shouldRetry(err error, resp *http.Response, numRetries int) bool {
+	if numRetries >= s.MaxNetworkRetries {
+		return false
+	}
+
+	if err != nil {
+		return true
+	}
+
+	if resp.StatusCode == http.StatusConflict {
+		return true
+	}
+	return false
+}
+
+//sleepTime calculates sleeping/delay time in milliseconds between failure and a new one request.
+func (s *BackendConfiguration) sleepTime(numRetries int) (delay int) {
+	// Apply exponential backoff with minNetworkRetryDelay on the
+	// number of num_retries so far as inputs.
+	delay = minNetworkRetryDelay + minNetworkRetryDelay*numRetries*numRetries
+
+	//Do not allow the number to exceed maxNetworkRetryDelay.
+	if delay > maxNetworkRetryDelay {
+		delay = maxNetworkRetryDelay
+	}
+
+	//Apply some jitter by randomizing the value in the range of 75%-100%.
+	jitter := rand.Intn(delay / 4)
```

Comment:
Also looks like a bit of deviation from the stripe-ruby implementation. It's not the end of the world, but is there any particular reason to choose `4` here?

Reply:
In my opinion stripe-ruby value (`2`) is too big. Delay should increase smoothlyIn ruby implementation previous
 value could be greater then next. For example possible answer in ruby is:  `6939
15189
18939
26439
24439`.  So on 4-th iteration value is bigger then on 5-th.   Anyway if you want, I will change it - it's not the end of the world :)

Reply:
Yeah, it's not a big deal. I was just curious where the `4` in particular came from.